### PR TITLE
Fix ChOp GPU builds

### PR DESCRIPTION
**Pull request by DanilaFe**
Recently, ChOp seems to have enabled the use of some CUDA-based GPU code that is _not_ generated by Chapel. This means we need to build the CUDA libraries too, and link against them when running ChOp testing. This also includes a patch to make the code work (which it seems to not on `main`, because `GPU_device_count` is used but not declared)

### CODE REVIEW

File: test/gpu/native/studies/chop/patches/GPU_aux.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@@ -0,0 +1,9 @@
++module GPU_aux{
++
++	use CTypes;
++
++    require "headers/GPU_aux.h";
++
++    extern proc GPU_device_count(): c_int;
++
++}
```
Comment (e-kayrakli) [inline test/gpu/native/studies/chop/patches/GPU_aux.patch:15]: Are we adding this file in a patch?

Was there any major obstacle in dumping the contents here into `queens_GPU_single_locale.chpl`?

**Review by e-kayrakli — APPROVED** [review]
I'm OK with this. An ideal solution is probably to work with Tiago to sort this out in a nicer way, but let's move on for now. (ie. "why need `GPU_device_count`, where you can `here.gpus.size`?)

The good part about this is that it now uses both interop and native GPU programming at the same time.

File: test/gpu/native/studies/chop/patches/GPU_aux.patch (hunk 1 above)
Comment (DanilaFe) [reply test/gpu/native/studies/chop/patches/GPU_aux.patch:15]: We are adding. No major obstacle, I'm just copying (what appears to me to be) the structure of "`other_modules` contains a copy of `modules` with modifications", including `GPU_Aux.chpl`)

**Comment by DanilaFe** [comment]
I think that it's nice that this now tests `extern` CUDA procs together with Chapel-generated kernels.

**Comment by DanilaFe** [comment]
This was fixed upstream, so I'll close the PR for the time being.

**Pull request closed without merging**